### DISCUSSION TITLE
Anyway to run adb commands using this

### DISCUSSION THREAD

**Discussion by waleedzafar68** [yume-chan/ya-webadb#547] [Q&A] [answered]
Hey @yume-chan So I know the shell basically runs adb she'll <command>. Is there a way to run adb backup and other such commands. Developing a feature for Android forensics. Correct me if i am wrong

**Comment by yume-chan** [comment, marked as answer]
Every ADB command needs be re-implemented in code. Except `adb backup` just executes `bu` on device and pipe the output (https://github.com/yume-chan/ya-webadb/issues/354#issuecomment-1011234169) so you can use `bu` directly.

**Reply by waleedzafar68** [reply]
Thank you so much. Just another question, can you point me to the section of the directory structure where you have the code for the adb commands

**Reply by yume-chan** [reply]
https://github.com/yume-chan/ya-webadb/tree/main/libraries/adb/src/commands

**Comment by waleedzafar68** [comment]
Thank you so very much man!

**Comment by waleedzafar68** [comment]
Sorry man for bugging you again @yume-chan but Is the download functionality not working like cannot download any file or recording

**Reply by yume-chan** [reply]
Do you mean it doesn't work in my online demo? Or your own project? If the later, how did you do that?

**Reply by waleedzafar68** [reply]
Yes in your online demo and sometimes locally as well. Giving error 404

**Reply by yume-chan** [reply]
404 means the service worker failed to register. Can you confirm it's happening on https://ya-webadb.vercel.app/ or other my own hosted environments? Because the most common cause is lacking of HTTPS, I have never seen that on my own domains. https://github.com/yume-chan/ya-webadb/blob/main/CONTRIBUTING.md#1-webusb-and-file-downloading-doesnt-work-in-developmentself-host-environment

**Comment by waleedzafar68** [comment]
I am actually using this as a base for an over-the-network Android Forensics

**Reply by yume-chan** [reply]
That's interesting! Can you talk about how do you connect to Android devices?

**Reply by waleedzafar68** [reply]
Sorry, for the late reply but download file not working on your domains as welll. ![IMG_20230501_161136.jpg](https://user-images.githubusercontent.com/91959220/235445915-5fe1c9c4-cbee-4732-97fb-03b905be1a98.jpg)

![IMG_20230501_161145.jpg](https://user-images.githubusercontent.com/91959220/235445916-e55178d0-bc5a-4053-aeb2-bc189a5130e9.jpg)

**Reply by yume-chan** [reply]
It uses a Service Worker that generates HTTP response dynamically to emulate file downloads. Your files can't possibly be on my server, so third-party download managers will only get 404.
